spec/apps/happy-active-when/happy-active-when.spec.js

```diff
@@ -0,0 +1,80 @@
+import * as singleSpa from "single-spa";
+
+describe(`happy-active-when`, () => {
+  let myApp;
+  const realLocation = global.location;
+
+  beforeAll(() => {
+    singleSpa.registerApplication({
+      name: "myApp",
+      app: () => import("./happy-active-when.app.js"),
+      activeWhen: [
+        "/#/appWithRegularPrefix",
+        "/pathname",
+        (location) => location.pathname === "/specificCriteria",
+        "/#/hashResource/:id/hashSubResource/:hashSubResourceId",
+        "/resource/:id/subresource/:subId",
+      ],
+    });
+  });
+
+  beforeEach(() => {
+    return import("./happy-active-when.app.js")
+      .then((app) => (myApp = app))
+      .then((app) => app.reset());
+  });
+
+  afterEach(() => {
+    global.location = realLocation;
+  });
+
+  it(`goes through the whole lifecycle successfully on multiple activeWhen conditions`, async () => {
+    singleSpa.start();
+    expect(myApp.isMounted()).toEqual(false);
+    expect(singleSpa.getMountedApps()).toEqual([]);
+    newLocation("http://mock.com/pathname");
+    await singleSpa.triggerAppChange();
+    expectMyAppToBeMounted();
+    newLocation("http://mock.com/#/unregisteredPath");
+    await singleSpa.triggerAppChange();
+    expectMyAppToBeUnMmounted();
+    newLocation("http://mock.com/#/appWithRegularPrefix");
+    await singleSpa.triggerAppChange();
+    expectMyAppToBeMounted();
+    newLocation("http://mock.com/#/unregisteredPath");
+    await singleSpa.triggerAppChange();
+    expectMyAppToBeUnMmounted();
+    newLocation("http://mock.com/specificCriteria");
+    await singleSpa.triggerAppChange();
+    expectMyAppToBeMounted();
+    newLocation("http://mock.com/#/unregisteredPath");
+    await singleSpa.triggerAppChange();
+    expectMyAppToBeUnMmounted();
+    newLocation("http://mock.com/resource/1/subresource/1");
+    await singleSpa.triggerAppChange();
+    expectMyAppToBeMounted();
+    newLocation("http://mock.com/#/unregisteredPath");
+    await singleSpa.triggerAppChange();
+    expectMyAppToBeUnMmounted();
+    newLocation("http://mock.com/#/hashResource/1/hashSubResource/1");
```

Comment:
An important test to add (in general, just anywhere we can) is when the route matches the string but has a trailing slash at the end of it.